### DISCUSSION TITLE
use config.twitch auth key and disable-ads at the same time?

### DISCUSSION THREAD

**Discussion by Dan-Y89** [streamlink/streamlink#5749] [Q&A] [answered]
### Description

**Hi, it's not about how a feature works, rather a general question.**

I have just created my config.twitch file so that I don't have to enter the auth key all the time.
So I created the file as it is in the CLI:
twitch-api-header=Authorization=OAuth KEYabc123
Save in config.twitch. Now it says "Type: TWITCH File" in Windows, I hope this is correct.
Now I enter this in the Windows CMD as an example: 
streamlink --twitch-disable-ads --config twitch -o "C:\Users\m-say\Documents\TOOLS\CMD\xREC\boesesjuju0autchtest.mp4" twitch.tv/boesesjuju best

Now to the question: Can I use this?
Because when I use twitch-disable-ads, it says in the CMD: 
Will skip ad segments
only because I am sub in the channel, I will not get any ads.
Therefore, I ask, is this possible or do you always have to check first whether you are a sub in the desired channel or not and then specify accordingly whether you want to use the twitch-disable-ads function?

Thank you very much

**Comment by bastimeyer** [comment, marked as answer]
> Now it says "Type: TWITCH File" in Windows

Because your OS / file browser sees the `.twitch` file extension, which is arbitrary, just like **any other file** on your file system.

----

> `--config twitch`

This doesn't make any sense. [`--config=FILENAME`](https://streamlink.github.io/cli.html#cmdoption-config) loads a custom config file from the specified path.

Unless you've set `--no-config`, Streamlink loads config files in the following order: first the regular config file, then the plugin-specific config file (with the `.pluginname` file extension), and then any custom config file set using `--config`. Options get overridden in order, with arguments set on the CLI having the highest priority.

https://streamlink.github.io/cli/config.html

----

Check the docs and see what `--twitch-disable-ads` actually does:

- https://streamlink.github.io/cli/plugins/twitch.html#embedded-ads
- https://streamlink.github.io/cli.html#cmdoption-twitch-disable-ads

It simply **skips the ad HLS segments in Streamlink's output**, so the stream discontinuity which will happen (even when filtering out ads) can be most likely recovered from, compared to when not filtering out ads.

This doesn't "disable" ads on Twitch's side. They decide when to embed ads or other content, even if you're logged in and subscribed to a specific channel or subscribed to Twitch turbo / Amazon prime, or even if you're accessing the Twitch API or the HLS playlist URL from a different country (which they currently don't do though).
